modules/claims_api/spec/sidekiq/report_hourly_unsuccessful_submissions_spec.rb

```diff
@@ -80,24 +82,79 @@
-              expect(expected_absent_values.flatten).not_to include(arg)
-            end
-          end
-        end
-        # rubocop:enable RSpec/SubjectStub
+        expected_results(expected_vagov_claims, expected_absent_values, expected_present_values)
 
         subject.perform
       end
+
+      it 'does not alert for claims with specific errors' do
+        allow_any_instance_of(Flipper).to receive(:enabled?).with(:claims_hourly_slack_error_report_enabled)
+                                                            .and_return(true)
+
+        claim_one = FactoryBot.create(:auto_established_claim_va_gov, :errored, created_at: Time.zone.now,
+                                                                                transaction_id: 'transaction_1',
+                                                                                id: '1')
+        claim_two = FactoryBot.create(:auto_established_claim_va_gov, :errored, created_at: 2.hours.ago,
+                                                                                transaction_id: 'transaction_1',
+                                                                                id: '2')
+        claim_three = FactoryBot.create(:auto_established_claim_va_gov, :errored, created_at: Time.zone.now,
+                                                                                  transaction_id: 'transaction_2',
+                                                                                  id: '3')
+        claim_four = FactoryBot.create(:auto_established_claim_va_gov, :errored, created_at: Time.zone.now,
+                                                                                 transaction_id: 'transaction_3',
+                                                                                 id: '4')
+
+        FactoryBot.create(:auto_established_claim_va_gov,
+                          :errored,
+                          created_at: 30.seconds.ago,
+                          evss_response: "[{'status'=>'422', " \
+                                         "'title'=>'Backend Service Exception', " \
+                                         "'detail'=>'The Maximum number of EP codes have been " \
+                                         "reached for this benefit type claim code'}]",
+                          transaction_id: 'transaction_4')
+
+        FactoryBot.create(:auto_established_claim_va_gov,
+                          :errored,
+                          created_at: 120.seconds.ago,
+                          evss_response: "[{'status'=>'422', " \
+                                         "'title'=>'Backend Service Exception', " \
+                                         "'detail'=>'Claim could not be established. Retries will fail.'}]",
+                          transaction_id: 'transaction_5')
+
+        expected_vagov_claims = [claim_three.id, claim_four.id]
+        expected_absent_values = [claim_one.id, claim_two.id]
```

Comment:
Are we using these two claims anywhere in the test? i.e., are we actually testing that the max EP codes and retry errors are filtered out?

Reply:
When that job runs it gathers all those claims, sends them to the `ReportHourlyUnsuccessfulSubmissions` `perform`.  They should get filtered out in the `get_filtered_unique_errors`. As you can see from the images I believe this test is solid and testing what I expected it to, unless you are getting something else?

Image 1: The four errored claims that make it here after the unique transaction_ids get filtered out
Image 2: When the no retry errored claims are filtered out we are left with two
Image 3: The 4 claims in `unique_errors` you can see the two errored no retry because they have `evss_Response` data (their IDs will not be included next)
Image 4: IDs of the 2 left over claims
Images 5 & 6: The claims with matching IDs in filtered_errors

* 1
<img width="455" alt="Screenshot 2024-11-06 at 12 23 41 PM" src="https://github.com/user-attachments/assets/4551e96d-90a6-4776-a3d3-c43e4ff95ed2">

* 2
<img width="487" alt="Screenshot 2024-11-06 at 12 23 59 PM" src="https://github.com/user-attachments/assets/478b84bc-193c-4bbf-aa2d-4b3201fa59a1">

* 3
<img width="1058" alt="Screenshot 2024-11-06 at 12 24 53 PM" src="https://github.com/user-attachments/assets/7e9e9771-7ca9-4922-9931-daf8d91fd246">

* 4
<img width="565" alt="Screenshot 2024-11-06 at 12 26 03 PM" src="https://github.com/user-attachments/assets/0e08b821-9f18-48f6-93f6-7c5ac9d40cd3">

* 5 & 6
<img width="1071" alt="Screenshot 2024-11-06 at 12 25 58 PM" src="https://github.com/user-attachments/assets/3193d3c1-a9dd-4902-8a22-65bc74835aaf">
<img width="1045" alt="Screenshot 2024-11-06 at 12 25 16 PM" src="https://github.com/user-attachments/assets/f8df42e2-6536-41c3-829d-e24cbf3eb2a6">

Reply:
Ah yep, the unassigned claims are being tested. I was thrown off by their lack of variable names, but the test does work as intended.